stdlib/Distributed/src/cluster.jl

```diff
@@ -603,7 +624,17 @@ function create_worker(manager, wconfig)
 
     # Start a new task to handle inbound messages from connected worker in master.
     # Also calls `wait_connected` on TCP streams.
-    process_messages(w.r_stream, w.w_stream, false)
+    procmsg_task = process_messages(w.r_stream, w.w_stream, false)
+    timeout = worker_timeout()
+    Timer(1, interval=1) do timer
+        timeout -= 1
+        if timeout <= 0.0
+            put!(rr_ntfy_join, :TIMEDOUT)
```

Comment:
This seems it could fail

Reply:
Yes, should handle it and close timer

Reply:
done in https://github.com/JuliaLang/julia/pull/32290/commits/c7ceaea79ab73a9e1f6db7bd7f2dcbada4c4787c